python/paddle/fluid/contrib/slim/quantization/qat2_int8_mkldnn_pass.py

```diff
@@ -74,7 +77,13 @@ def apply(self, graph):
         assert isinstance(graph,
                           IrGraph), 'graph must be the instance of IrGraph.'
 
-        graph = self._gather_scales(graph)
+        graph = self._gather_weight_scales_from_fake(graph)
+        if self._scale_from_attr:
+            graph = self._gather_scales_from_attr(graph)
+            graph = self._gather_scales_from_fake(graph)
+        else:
+            graph = self._gather_scales_from_fake(graph)
+            graph = self._gather_scales_from_attr(graph)
```

Comment:
By default, gather scales from attributes and then gather scales from fake ops, so the latter is at a higher priority.

It is necessary to check that the input and output scales are in _var_quant_scales for all quantized ops. If not, raise an error before applying quantization.

Reply:
@juncaipeng , do you mean the following procedure should be applied?
1. gather scales from all attributes,
2. gather scales from fake ops, override scales gathered in 1. if present.

Reply:
go through all quantized ops and assert scales all exists. Do not wait until we start quantization

Reply:
Yes. Just delete the _scale_from_attr.

Reply:
yes. Besides, delete the _scale_from_attr.

Reply:
Done.